src/lib.c

```diff
@@ -227,6 +240,119 @@ efi_variables_supported(void)
 	return 1;
 }
 
+static void
+search_esp_filename(const char *filename, char **esp_filename)
+{
+	size_t num_paths = sizeof(esp_paths) / sizeof(esp_paths[0]);
+	size_t esp_filename_size = 0;
+
+	for (size_t i = 0; i < num_paths; ++i) {
+		esp_filename_size = strlen(esp_paths[i]) + strlen(filename) + 2;
+		*esp_filename = malloc(esp_filename_size);
+		if (!*esp_filename) {
+			fprintf(stderr, "Error: failed to allocate memory for esp_filename\n");
+			exit(1);
+		}
+		snprintf(*esp_filename, esp_filename_size, "%s/%s", esp_paths[i], filename);
+
+		struct stat buffer;
+		if (stat(*esp_filename, &buffer) == 0) {
+			return;
+		}
+
+		free(*esp_filename);
+		*esp_filename = NULL;
+	}
+}
+
+static int
+get_esp_filename(char **esp_filename)
+{
+	size_t filename_size;
+	uint32_t filename_attr;
+	uint8_t *filename = NULL;
+	int rc = 0;
+
+	rc = efi_get_variable(GUID_RTSV, NAME_RTSV, &filename, &filename_size, &filename_attr);
+	if (rc < 0) {
+		fprintf(stderr, "efivar: Error %d getting filename from RTStorageVolatile\n", rc);
```

Comment:
Remove this print completely. If the variable is not there, it's probably a normal system that doesn't need to store variables in a file. 

If the variable is there and we fail to read an existing filename the code will fail later during efi_save_esp_filename()

Reply:
Replace the print with a code comment. Thanks